File: MainApp/Renting/src/main/java/jvn/Renting/model/RentReport.java
package jvn.Renting.model;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class RentReport {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private String comment;

    @OneToOne
    private RentInfo rentInfo;

    @Column
    private Double additionalCost;

    @Column
    private Integer madeMileage;

    @Column
    private Boolean paid;
}

File: MainApp/AgentApp/Frontend/src/app/model/rentReport.ts
import { RentInfo } from "./rentInfo";

export class RentReport {
  id: number;
  comment: string;
  // rentInfo: RentInfo;
  additionalCost: number;
  madeMileage: number;

  constructor(
    // rentInfo: RentInfo,
    madeMileage: number,
    comment?: string,
    additionalCost?: number,
    id?: number
  ) {
    // this.rentInfo = rentInfo;
    this.madeMileage = madeMileage;
    this.additionalCost = additionalCost;
    this.comment = comment;
    this.id = id;
  }
}
